cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -1127,9 +1128,20 @@ private V1ResourceRequirements createResourceRequirements(Type resourceType, Res
       .addToRequests("memory", new Quantity(String.format("%dMi", memoryToRequest)));
 
     if (resourceType == V1Job.class) {
+      int cpuLimit = resourceSpec.getVirtualCores() * 1000;
+      int memoryLimit = resourceSpec.getMemorySize();
+      if (cpuLimit < cpuToRequest) {
+        throw new RuntimeException(
+          String.format("CPU limits %d for kubernetes pod is lower than requested cpu %d", cpuLimit, cpuToRequest));
```

Comment:
if we're checking, it's better to do the check on the multiplier and say it should be between 0 and 1, since that's the action that the admin will need to take to correct things

Reply:
Ah I see.

Reply:
yea makes sense since we consider limit to be whatever user program has provided.